pkg/cmd/auth/setupgit/setupgit_test.go

```diff
@@ -22,15 +27,63 @@ func (gf *mockGitConfigurer) Setup(hostname, username, authToken string) error {
 	gf.hosts = append(gf.hosts, hostname)
 	return gf.setupErr
 }
+func TestNewCmdSetupGit(t *testing.T) {
+	tests := []struct {
+		name     string
+		cli      string
+		wantsErr bool
+		errMsg   string
+	}{
+		{
+			name:     "--force without hostname",
+			cli:      "--force",
+			wantsErr: true,
+			errMsg:   "`--force` must be used in conjunction with `--hostname`",
+		},
+		{
+			name:     "no error when --force used with hostname",
+			cli:      "--force --hostname ghe.io",
+			wantsErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			f := &cmdutil.Factory{}
+
+			argv, err := shlex.Split(tt.cli)
+			require.NoError(t, err)
+
+			cmd := NewCmdSetupGit(f, func(opts *SetupGitOptions) error {
+				return nil
+			})
 
+			// TODO cobra hack-around
+			cmd.Flags().BoolP("help", "x", false, "")
+
+			cmd.SetArgs(argv)
+			cmd.SetIn(&bytes.Buffer{})
+			cmd.SetOut(io.Discard)
+			cmd.SetErr(io.Discard)
+
+			_, err = cmd.ExecuteC()
+			if tt.wantsErr {
+				require.Error(t, err)
+				require.Equal(t, err.Error(), tt.errMsg)
+				return
+			}
+
+			require.NoError(t, err)
+		})
+	}
+}
 func Test_setupGitRun(t *testing.T) {
 	tests := []struct {
 		name               string
 		opts               *SetupGitOptions
 		setupErr           error
 		cfgStubs           func(*testing.T, config.Config)
 		expectedHostsSetup []string
-		expectedErr        string
+		expectedErr        error
```

Comment:
We're testing the return from a function, so I'd rather use the real type rather than the string form of that type like we might test in the stdout.